src/Microsoft.AspNet.Mvc.Razor/Compilation/DnxRoslynCompilationService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Reflection.PortableExecutable;
+using System.IO;
+using Microsoft.CodeAnalysis;
+using Microsoft.Dnx.Compilation.CSharp;
+using Microsoft.Extensions.CompilationAbstractions;
+using Microsoft.Extensions.PlatformAbstractions;
+using Microsoft.Extensions.OptionsModel;
+using System.Runtime.Versioning;
+
+namespace Microsoft.AspNet.Mvc.Razor.Compilation
+{
+    public class DnxCompilationCallback
+    {
+        private readonly Lazy<List<MetadataReference>> _applicationReferences;
+        private readonly ILibraryExporter _libraryExporter;
+        private readonly IApplicationEnvironment _environment;
+        private readonly ICompilerOptionsProvider _compilerOptionsProvider;
+        private readonly ConcurrentDictionary<string, AssemblyMetadata> _metadataFileCache =
+                new ConcurrentDictionary<string, AssemblyMetadata>(StringComparer.OrdinalIgnoreCase);
+        private RazorViewEngineOptions _options;
+
+        /// <summary>
+        /// Initalizes a new instance of the <see cref="RoslynCompilationService"/> class.
+        /// </summary>
+        /// <param name="environment">The environment for the executing application.</param>
+        /// <param name="loaderAccessor">
+        /// The accessor for the <see cref="IAssemblyLoadContext"/> used to load compiled assemblies.
+        /// </param>
+        /// <param name="libraryManager">The library manager that provides export and reference information.</param>
+        /// <param name="compilerOptionsProvider">
+        /// The <see cref="ICompilerOptionsProvider"/> that provides Roslyn compilation settings.
+        /// </param>
+        /// <param name="host">The <see cref="IMvcRazorHost"/> that was used to generate the code.</param>
+        public DnxCompilationCallback(
```

Comment:
Because i want to isolate library manger and `ICompilerOptionsProvider` from RoslynCompilationServices so they are not required or i still would need to write some kind of shims for them to make MVC work on dotnet